src/main/scala/tests.scala

```diff
@@ -43,11 +43,11 @@ import Node._
 object equivRecFN
 {
     def apply(expWidth: Int, sigWidth: Int, a: Bits, b: Bits) = {
-        val top4A = a(expWidth + sigWidth, expWidth + sigWidth - 3)
-        val top4B = b(expWidth + sigWidth, expWidth + sigWidth - 3)
+        val top4A = a(expWidth + sigWidth, expWidth + sigWidth - 3).toUInt
```

Comment:
Should the function args be changed to UInt instead of Bits, avoiding these casts?